File: LeetCode/0263. Ugly Number/solution.py
"""
1013 / 1013 test cases passed.
Runtime: 44 ms
Memory Usage: 15 MB
"""
class Solution:
    def isUgly(self, n: int) -> bool:
        if n <= 0:
            return False
        while not n % 5:
            n /= 5
        while not n % 3:
            n /= 3
        while not n % 2:
            n /= 2
        return n == 1

File: LeetCode/0263. Ugly Number/solution.cpp
/**
 * 1013 / 1013 test cases passed.
 * Runtime: 4 ms
 * Memory Usage: 5.8 MB 
 */
class Solution {
public:
    bool isUgly(int n) {
        if (n <= 0) return false;
        while (n % 5 == 0) n /= 5;
        while (n % 3 == 0) n /= 3;
        while (n % 2 == 0) n /= 2;
        return n == 1;
    }
};
